### DISCUSSION TITLE
Pin numbers of I2C interface

### DISCUSSION THREAD

**Discussion by Hristo-Nikodimov-Nenkov** [micropython/micropython#11319] [Hardware & Peripherals]
Greetings.
I have  RP Pico like this 
[![image](https://user-images.githubusercontent.com/36823393/233855131-fa8ff336-9304-4f4e-9a55-2b334c4d70e7.png)](url)

I'll use ssd1306 screen and plan to conect it to the I2C on the bottom. I need the Pin numbers of SWDIO and SWCLK.

**Comment by GitHubsSilverBullet** [comment]
You do realize that you're talking about the **S**erial **W**ire **D**ebug interface?
The pins are the same for every RP2040, so it' SWCLK 24, SWDIO 25.

**Reply by Hristo-Nikodimov-Nenkov** [reply]
This model has LED on Pin 25 ( left of the Type C connector ).

**Reply by robert-hh** [reply]
SWCLK and SWDIO are on the chip package Pins 24 and 25, not GPIO24 and GPIO25. I could not tell from the datasheet, that SWCLK or SWDIO can be user for a different purpose. Looking at the address space, if could be GPIO30 and GPIO31. But you may have to configure the DBGFORCE Register as well.

Edit: 30 and 31 are not accepted as pin number by machine.Pin().

**Reply by Hristo-Nikodimov-Nenkov** [reply]
Thanks.

**Comment by peterhinch** [comment]
To connect to an SSD1306 you need I2C which is available on numerous pins, e.g. GP16 and GP17. Ignore the SWD interface unless you have a specific application for it.

Don't forget the pullup resistors.

**Reply by Hristo-Nikodimov-Nenkov** [reply]
I intend to use 3.3V and GND from the debug and 14, 15 or 15, 16 for I2C and something between 5KOhm and 10KOhm for pullup.
